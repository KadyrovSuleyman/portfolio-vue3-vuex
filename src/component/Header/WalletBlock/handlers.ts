import { ComputedRef } from 'vue';
import { StateT } from './adapter';

const createShowConnectWalletModal = (state: ComputedRef<StateT>) => () => {
  state.value.sendRequest();
};

export default createShowConnectWalletModal;
