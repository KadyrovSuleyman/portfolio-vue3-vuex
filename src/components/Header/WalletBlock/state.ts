import { Store } from 'vuex';

export interface StateT {
  isWalletConnect: boolean,
  sendRequest: () => Promise<any>,
}

export const adaptState = (store: Store<any>): StateT => ({
  isWalletConnect: store.getters['wallet/isConnected'],
  sendRequest: () => store.dispatch('waiting/connectWallet'),
});
