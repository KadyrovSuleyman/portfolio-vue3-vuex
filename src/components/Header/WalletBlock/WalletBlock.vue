<script setup lang="ts">

import { computed } from 'vue';
import Div from '@/elements/Div/Div.vue';
import propsObj from '@/elements/propsObj';
import { useStore } from 'vuex';
import WalletDiv from './WalletDiv/WalletDiv.vue';
import ConnectBlock from './ConnectBlock/ConnectBlock.vue';
import { adaptState } from './state';
import createShowConnectWalletModal from './handlers';

// eslint-disable-next-line no-undef
const props = defineProps({ ...propsObj });
const comp = computed(() => ({ elem: props.elem || 'walletBlock' }));

const store = useStore();
const state = computed(() => adaptState(store));

const showConnectWalletModal = createShowConnectWalletModal(state);

</script>

<template>
  <Div :block="props.block" :elem="comp.elem" :mods="props.mods">
    <WalletDiv :block="comp.elem" v-if="state.isWalletConnect" />
    <ConnectBlock :block="comp.elem" v-else
      :onClick="showConnectWalletModal"
    />
  </Div>
</template>

<style lang="scss">
  @use 'styles' as *;
</style>
