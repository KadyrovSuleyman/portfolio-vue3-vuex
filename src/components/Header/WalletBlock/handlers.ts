import { ComputedRef } from 'vue';
import { StateT } from './state';

const createShowConnectWalletModal = (state: ComputedRef<StateT>) => () => {
  state.value.sendRequest();
};

export default createShowConnectWalletModal;
