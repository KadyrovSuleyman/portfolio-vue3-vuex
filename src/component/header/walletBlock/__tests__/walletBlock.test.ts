import { mount } from '@vue/test-utils';
import { computed, ref } from 'vue';
import { createStore, Store } from 'vuex';
import WalletBlock from '../walletBlock.vue';

let wrapper = mount(WalletBlock, { global: { plugins: [createStore({})] } });
wrapper.unmount();
afterEach(() => {
  wrapper.unmount();
});

it('walletBlock renders', () => {
  wrapper = mount(WalletBlock, { global: { plugins: [createStore({})] } });

  expect(wrapper.find('div').classes()).toEqual(['walletBlock']);
});

it('watchs props changes', async () => {
  const Div = {
    props: [],

    setup() {
      const isSelected = ref(false);
      const select = () => {
        isSelected.value = !isSelected.value;
      };
      const mods = computed(() => ({ selected: isSelected.value }));

      return {
        isSelected,
        select,
        mods,
      };
    },
    components: {
      WalletBlock,
    },

    template: `
      <div class='root'>
        <WalletBlock :mods="mods"/>
        <button class="test-btn" @click="select"></button>
      </div>
    `,
  };
  const wr = mount(Div, { global: { plugins: [createStore({})] } });
  expect(wr.find('.walletBlock').classes()).toEqual(['walletBlock']);

  await wr.find('.test-btn').trigger('click');
  expect(wr.find('.walletBlock').classes()).toEqual(['walletBlock', 'walletBlock__selected']);

  await wr.find('.test-btn').trigger('click');
  expect(wr.find('.walletBlock').classes()).toEqual(['walletBlock']);

  wr.unmount();
});

// =========================================
jest.mock('../adapter', () => ({
  __esModule: true,
  default: (store: Store<any>) => ({
    isWalletConnect: store.state.connect,
  }),
}));

it('walletBlock if wallet connected, connectBlock if dont', async () => {
  const store = createStore({
    state: {
      connect: false,
    },
    mutations: {
      changeConnection: (state) => { state.connect = !state.connect; },
    },
  });

  wrapper = mount(WalletBlock, {
    global: {
      plugins: [store],
    },
  });

  expect(wrapper.find('.walletBlock-walletDiv').exists()).toBeFalsy();
  expect(wrapper.find('.walletBlock-connectBlock').exists()).toBeTruthy();

  store.commit('changeConnection');
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.walletBlock-walletDiv').exists()).toBeTruthy();
  expect(wrapper.find('.walletBlock-connectBlock').exists()).toBeFalsy();

  store.commit('changeConnection');
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.walletBlock-walletDiv').exists()).toBeFalsy();
  expect(wrapper.find('.walletBlock-connectBlock').exists()).toBeTruthy();
});
