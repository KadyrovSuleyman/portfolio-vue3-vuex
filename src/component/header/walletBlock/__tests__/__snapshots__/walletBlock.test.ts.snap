// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`walletBlock if wallet connected, connectBlock if dont 1`] = `
<div class="walletBlock">
  <div class="walletBlock-walletDiv">
    <div class="walletDiv-addressBlock"><span class="addressBlock-span"></span><button class="addressBlock-button addressBlock-button__icon_copy"></button></div><span class="walletDiv-span"> </span><button class="walletDiv-button walletDiv-button__icon_coin"><img></button><button class="walletDiv-button walletDiv-button__icon_chevron"></button>
  </div>
</div>
`;

exports[`walletBlock if wallet connected, connectBlock if dont 2`] = `
<div class="walletBlock">
  <div class="walletBlock-connectBlock"><button class="connectBlock-button"> Connect wallet </button></div>
</div>
`;

exports[`walletBlock renders 1`] = `
<div class="walletBlock">
  <div class="walletBlock-connectBlock"><button class="connectBlock-button"> Connect wallet </button></div>
</div>
`;
